### PR TITLE
Remove AccountTotpPage from the testsuite #15201

### LINKED ISSUE #15201: Remove AccountTotpPage from the testsuite
### Description

Tests that are not testing the account console should not rely on it. This task is to remove [AccountTotpPage](https://github.com/keycloak/keycloak/blob/bfce612641a70e106b20b136431f0e4046b5c37f/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/AccountTotpPage.java) and usage of it, re-writing to use REST APIs as applicable.

### PR DESCRIPTION
**Pull request by lhanusov**
fix #15201

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/storage/BackwardsCompatibilityUserStorageTest.java [hunk 1]
```diff
@@ -171,58 +169,55 @@ public void testOTPUpdateAndLogin() {
...
+
+        testAppHelper.logout();
 
         // Authenticate as the user with bad OTP
-        loginPage.login("otp1", "pass");
+        testAppHelper.startLogin("otp1", "pass");
         loginTotpPage.assertCurrent();
         loginTotpPage.login("7123456");
-        assertCurrentUrlDoesntStartWith(testRealmAccountPage);
+        assertTrue(loginTotpPage.isCurrent());
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/storage/BackwardsCompatibilityUserStorageTest.java:187]: `loginTotpPage.assertCurrent()` is slightly better?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/storage/BackwardsCompatibilityUserStorageTest.java [hunk 2]
```diff
@@ -171,58 +169,55 @@ public void testOTPUpdateAndLogin() {
...
         Assert.assertNotNull(loginTotpPage.getInputError());
 
         // Authenticate as the user with correct OTP
         loginTotpPage.login(totp.generateTOTP(totpSecret));
-        assertCurrentUrlStartsWith(testRealmAccountPage);
-        testRealmAccountPage.logOut();
+        testAppHelper.completeLogin();
+        assertTrue(appPage.isCurrent());
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/storage/BackwardsCompatibilityUserStorageTest.java:193]: `appPage.assertCurrent()` is slightly better

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/storage/BackwardsCompatibilityUserStorageTest.java [hunk 3]
```diff
@@ -266,29 +261,32 @@ public void testSearchUserStorage() {
...
         // Login as the user and setup OTP
-        testRealmAccountPage.navigateTo();
-        loginPage.login("otp1", "pass");
+        testAppHelper.startLogin("otp1", "pass");
 
         configureTotpRequiredActionPage.assertCurrent();
         String totpSecret = configureTotpRequiredActionPage.getTotpSecret();
         configureTotpRequiredActionPage.configure(totp.generateTOTP(totpSecret));
-        assertCurrentUrlStartsWith(testRealmAccountPage);
+        assertTrue(appPage.isCurrent());
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/storage/BackwardsCompatibilityUserStorageTest.java:284]: Same as above

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/ResetCredentialsAlternativeFlowsTest.java [hunk 4]
```diff
@@ -358,14 +345,17 @@ public void resetCredentialsVerifyCustomOtpLabelSetProperly() {
...
             OAuthClient.AccessTokenResponse response = oauth.doAccessTokenRequest(code, "password");
 
-            accountTotpPage.open();
-            Assert.assertTrue(accountTotpPage.isCurrent());
             String customOtpLabel = "my-original-otp-label";
-            accountTotpPage.configure(totp.generateTOTP(accountTotpPage.getTotpSecret()), customOtpLabel);
+
+            // Setup OTP
+            doAIA();
+            Assert.assertTrue(totpPage.isCurrent());
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/ResetCredentialsAlternativeFlowsTest.java:356]: totpPage.assertCurrent

**Review by mposolda — CHANGES_REQUESTED** [review]
@lhanusov Thanks!

This PR will require some rebase due the conflict with other PR for consents https://github.com/keycloak/keycloak/pull/17651 .

Besides that, I've added some inline comments (basically I suggested to use something like `fooPage.assertCurrent()` instead of `assertTrue(fooPage.isCurrent())` (See method `assertCurrent()` for the details).

**Comment by lhanusov** [comment]
@mposolda Hello Marek, thank you for the review, suggestions resolved and applied.

**Review by mposolda — APPROVED** [review]

**Pull request merged**